src/web/json_string.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2011-2017 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <bitcoin/server/web/json_string.hpp>
+
+// Explicitly use std::placeholders before the json_parser is included
+// to avoid boost not using specified placeholders in the json_parsing
+// helper classes.
+#include <functional>
+using namespace std::placeholders;
+
```

Comment:
I think the above is a little better.  If not this, I think the fully commented version would look something like this instead:

```
#ifdef BOOST_BIND_NO_PLACEHOLDERS
// If we're not using boost::placeholders, explicitly use the                                                                                                                                                                                                                             
// std::placeholders here for usage internally to the boost parsing                                                                                                                                                                                                                       
// helpers included from json_parser.hpp                                                                                                                                                                                                                                                  
#include <functional>
using namespace std::placeholders;
#else
// Otherwise, explicitly use boost::placeholders before the                                                                                                                                                                                                                               
// json_parser is included.                                                                                                                                                                                                                                                               
#include <boost/bind/placeholders.hpp>
using namespace boost::placeholders;
#endif // BOOST_BIND_NO_PLACEHOLDERS   
```

Reply:
This is better, but this is a pervasive issue that should be generally resolvable using the boost define. Do you know why that isn’t working? A lot of other code relies on it.

Reply:
@evoskuil I think it's confusing 2 issues.  The boost define is to make sure there's no ambiguity between boost::placeholders and std::placeholders.  When used/defined, it does not include boost/placeholders.hpp, so that code already using std::placeholders will work properly.  Our code uses _neither_ boost::placeholders nor std::placeholders at the time of include, so this is the other issue -- we need to define one.  Earlier in the code, we already asked boost to _not_ include boost::placeholders, so our choices (given that the json_parser requires one to be defined) are to 1) use std::placeholders, or 2) use boost::placeholders.  You can see the #ifdef in the comment doing just that depending on if the boost define is used or not, and either of those situations works.